datafusion/physical-expr-common/src/aggregate/groups_accumulator/prim_op.rs

```diff
@@ -102,9 +102,13 @@ where
             values,
             opt_filter,
             total_num_groups,
-            |group_index, new_value| {
-                let value = &mut self.values[group_index];
-                (self.prim_fn)(value, new_value);
+            |group_index, was_valid, new_value| {
```

Comment:
I don't think the existing implementation have to change, or do they?

Reply:
Perhaps not, although conceptually an accumulator is just a map-reduce-map via some monoid or semigroup. The current implementation supports only monoids (need an empty value) but it doesn't support semigroups (no empty value). We could use this for something like `FirstValue` or `AnyValue` but I guess we could also implement it specifically for that use case.

Reply:
We could even drop the initial value parameter since it doesn't matter in this case.